src/lib/grid-list/tile-coordinator.ts

```diff
@@ -91,6 +91,8 @@ export class TileCoordinator {
       // If we've reached the end of the row, go to the next row.
       if (this.columnIndex + tileCols > this.tracker.length) {
         this._nextRow();
+        gapStartIndex = this.tracker.indexOf(0, this.columnIndex);
```

Comment:
I'm not sure whether this makes sense. What it does is that it looks for the value `0`, starting from the `columnIndex`. Since the array is filled with zeroes (see https://github.com/angular/material2/pull/12980/files#diff-ddccaadf0ea62282a345db62650d07f7R58), it'll always stop at the next match.

Reply:
The array is not always filled with 0.  In this test case, the row 3 and 4 will only have 0 in the cell 5 and 6. Which is the empty cell in that row.

Reply:
This line actually isn't going to do anything at all. `gapStartIndex` is re-written after the `continue`